MathLib/LinAlg/UnifiedMatrixSetters.cpp

```diff
@@ -83,25 +83,35 @@ double norm(PETScVector const& x)
 void setVector(PETScVector& v,
                std::initializer_list<double> values)
 {
-    (void) v; (void) values;
-    // TODO implement
-}
+    std::vector<double> const vals(values);
+    std::vector<PETScVector::IndexType> idcs(vals.size());
+    std::iota(idcs.begin(), idcs.end(), 0);
 
+    v.set(idcs, vals);
+}
 
 void setMatrix(PETScMatrix& m,
                PETScMatrix::IndexType const rows,
                PETScMatrix::IndexType const cols,
                std::initializer_list<double> values)
 {
-    (void) m; (void) rows; (void) cols; (void) values;
-    // TODO implement
+    m.setZero();
+    addToMatrix(m, rows, cols, values);
 }
 
 void setMatrix(PETScMatrix& m, Eigen::MatrixXd const& tmp)
 {
+    m.setZero();
+    std::vector<PETScVector::IndexType> rows(tmp.rows());
+    std::vector<PETScVector::IndexType> cols(tmp.cols());
+
+    std::iota(rows.begin(), rows.end(), 0);
+    std::iota(cols.begin(), cols.end(), 0);
+
+    // PETSc wants row-major
+    Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor> tmp_ = tmp;
```

Comment:
Maybe it is possible to check if `tmp` is already in `RowMajor`? Then the copy would not be necessary.